nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-label.js

```diff
@@ -527,7 +538,7 @@
 
             // update
             var updated = selection.merge(entered);
-            updated.call(updateLabels).call(nfCanvasUtils.position, transition);
+            updated.call(updateLabels).call(nfCanvasUtils.position, transition).call(sort);
```

Comment:
We also need to introduce a call to `sort` in `add`. When a new Label is created, it's z-index will be 0. If it is placed directly on top of another Label that already has an established z-index we need to ensure this is rendered below. By default, it will render on top because it was added to the DOM later. If you try this out, you'll see it renders on top until the canvas polls which triggers a call to `update` that will re-sort and move the new Label below.

Reply:
Thanks for pointing this out. Unfortunately, I still see the behavior your describing even after updating `add` at line 484 from
```
            // update
            updateLabels(selection.merge(entered));
```
to
```
            // update
            var updated = selection.merge(entered);
            updated.call(updateLabels).call(nfCanvasUtils.position, transition).call(sort);
```
Does anything look off to you about that block of code?

Reply:
@sardell At first glance, this looks ok. Have you verified that the value of the `zIndex` coming back in the creation response? I wonder if it could be comparing `undefined` with some number. Setting a break point in `sort` should help steer in the right direction.

Reply:
@mcgilman Turned out that `transition` wasn't defined in `add`. This should now be functioning correctly.